src/main/java/net/sf/jabref/model/entry/BibtexEntryType.java

```diff
@@ -82,23 +82,17 @@ void addAllRequired(String... fieldNames) {
         return optionalFields.stream().filter(field -> !isPrimary(field)).collect(Collectors.toList());
     }
 
-    public abstract boolean hasAllRequiredFields(BibtexEntry entry, BibtexDatabase database);
-
     public String[] getUtilityFields() {
```

Comment:
I think, this method is useful also in other cases. I remember that there was a pattern for that: data objects should not carry "extended" methods, but a visitor or something similar should be used. `BibtexEntryTypeUtil`?